### ISSUE TITLE
[No build ID Error] when build rpm package for libseccomp locally

### ISSUE THREAD

**Issue by hazelnutsgz** [seccomp/libseccomp#167]
## What is the intention?
Currently, I am hacking the `libseccomp` repo and want to build the libseccomp 2.3.1 rpm package locally on CentOS 7 (3.10.0-957.10.1).

## What is the error?
When running `rpmbuild -ba libseccomp.spec` on the `~/rpmbuild/SPECS/`;
I got the following error output:
```
make[1]: Leaving directory `/home/gshe/rpmbuild/BUILD/libseccomp-2.3.1'
+ rm -f /home/gshe/rpmbuild/BUILDROOT/libseccomp-2.3.1-3.el7.x86_64//usr/lib64/libseccomp.la
+ rm -f /home/gshe/rpmbuild/BUILDROOT/libseccomp-2.3.1-3.el7.x86_64//usr/lib64/libseccomp.a
+ /usr/lib/rpm/find-debuginfo.sh --strict-build-id -m --run-dwz --dwz-low-mem-die-limit 10000000 --dwz-max-die-limit 110000000 /home/gshe/rpmbuild/BUILD/libseccomp-2.3.1
extracting debug info from /home/gshe/rpmbuild/BUILDROOT/libseccomp-2.3.1-3.el7.x86_64/usr/lib64/libseccomp.so.2.3.1
*** ERROR: No build ID note found in /home/gshe/rpmbuild/BUILDROOT/libseccomp-2.3.1-3.el7.x86_64/usr/lib64/libseccomp.so.2.3.1
error: Bad exit status from /var/tmp/rpm-tmp.SxDrlT (%install)


RPM build errors:
    Bad exit status from /var/tmp/rpm-tmp.SxDrlT (%install)

```

The file in `/var/tmp/rpm-tmp.SxDrlT ` is:
```
#!/bin/sh

  RPM_SOURCE_DIR="/home/gshe/rpmbuild/SOURCES"
  RPM_BUILD_DIR="/home/gshe/rpmbuild/BUILD"
  RPM_OPT_FLAGS="-O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches   -m64 -mtune=generic"
  RPM_LD_FLAGS="-Wl,-z,relro "
  RPM_ARCH="x86_64"
  RPM_OS="linux"
  export RPM_SOURCE_DIR RPM_BUILD_DIR RPM_OPT_FLAGS RPM_LD_FLAGS RPM_ARCH RPM_OS
  RPM_DOC_DIR="/usr/share/doc"
  export RPM_DOC_DIR
  RPM_PACKAGE_NAME="libseccomp"
  RPM_PACKAGE_VERSION="2.3.1"
  RPM_PACKAGE_RELEASE="3.el7"
  export RPM_PACKAGE_NAME RPM_PACKAGE_VERSION RPM_PACKAGE_RELEASE
  LANG=C
  export LANG
  unset CDPATH DISPLAY ||:
  RPM_BUILD_ROOT="/home/gshe/rpmbuild/BUILDROOT/libseccomp-2.3.1-3.el7.x86_64"
  export RPM_BUILD_ROOT

  PKG_CONFIG_PATH="${PKG_CONFIG_PATH}:/usr/lib64/pkgconfig:/usr/share/pkgconfig"
  export PKG_CONFIG_PATH

  set -x
  umask 022
  cd "/home/gshe/rpmbuild/BUILD"
    [ "$RPM_BUILD_ROOT" != "/" ] && rm -rf "${RPM_BUILD_ROOT}"
    mkdir -p `dirname "$RPM_BUILD_ROOT"`
    mkdir "$RPM_BUILD_ROOT"

cd 'libseccomp-2.3.1'

rm -rf "/home/gshe/rpmbuild/BUILDROOT/libseccomp-2.3.1-3.el7.x86_64"
mkdir -p "/home/gshe/rpmbuild/BUILDROOT/libseccomp-2.3.1-3.el7.x86_64//usr/lib64"
mkdir -p "/home/gshe/rpmbuild/BUILDROOT/libseccomp-2.3.1-3.el7.x86_64//usr/include"
mkdir -p "/home/gshe/rpmbuild/BUILDROOT/libseccomp-2.3.1-3.el7.x86_64//usr/share/man"
make V=1 DESTDIR="/home/gshe/rpmbuild/BUILDROOT/libseccomp-2.3.1-3.el7.x86_64" install
rm -f "/home/gshe/rpmbuild/BUILDROOT/libseccomp-2.3.1-3.el7.x86_64//usr/lib64/libseccomp.la"
rm -f "/home/gshe/rpmbuild/BUILDROOT/libseccomp-2.3.1-3.el7.x86_64//usr/lib64/libseccomp.a"



   /usr/lib/rpm/find-debuginfo.sh --strict-build-id -m --run-dwz\
   --dwz-low-mem-die-limit 10000000\
   --dwz-max-die-limit 110000000  "/home/gshe/rpmbuild/BUILD/libseccomp-2.3.1"

    /usr/lib/rpm/check-buildroot

    /usr/lib/rpm/redhat/brp-compress

    /usr/lib/rpm/redhat/brp-strip-static-archive /usr/bin/strip
    /usr/lib/rpm/brp-python-bytecompile /usr/bin/python 1
    /usr/lib/rpm/redhat/brp-python-hardlink
    /usr/lib/rpm/redhat/brp-java-repack-jar

```

#### built the folders 
`~/rpmbuild/BUILD/`
`~/rpmbuild/SOURCES/`
`~/rpmbuild/SRPMS/`
`~/rpmbuild/BUILDROOT/`
`~/rpmbuild/SPECS/`
`~/rpmbuild/RPMS/`

#### Downloaded Sources 
```
cd ~/rpmbuild/SOURCES
wget https://github.com/seccomp/libseccomp/releases/download/v2.3.1/libseccomp-2.3.1.tar.gz
```

#### SPEC file
Then I use the spec file **extracted from official libseccomp.x86_64 0:2.3.1-3.el7** in Centos
FYI:
My `libseccomp.spec`:
```
Summary: Enhanced seccomp library
Name: libseccomp
Version: 2.3.1
Release: 3%{?dist}
ExclusiveArch: %{ix86} x86_64 %{arm} aarch64 ppc ppc64 ppc64le s390 s390x
License: LGPLv2
Group: System Environment/Libraries
Source: https://github.com/seccomp/libseccomp/releases/download/v%{version}/%{name}-%{version}.tar.gz
## Source: /home/gshe/libseccomp_rpmbuild/%{name}-%{version}.tar.gz
URL: https://github.com/seccomp/libseccomp
%ifnarch s390
BuildRequires: valgrind
%endif

%description
The libseccomp library provides an easy to use interface to the Linux Kernel's
syscall filtering mechanism, seccomp.  The libseccomp API allows an application
to specify which syscalls, and optionally which syscall arguments, the
application is allowed to execute, all of which are enforced by the Linux
Kernel.

%package devel
Summary: Development files used to build applications with libseccomp support
Group: Development/Libraries
Requires: %{name}%{?_isa} = %{version}-%{release} pkgconfig

%description devel
The libseccomp library provides an easy to use interface to the Linux Kernel's
syscall filtering mechanism, seccomp.  The libseccomp API allows an application
to specify which syscalls, and optionally which syscall arguments, the
application is allowed to execute, all of which are enforced by the Linux
Kernel.

%prep
%setup -q

%build
%configure
make V=1 %{?_smp_mflags}

%install
rm -rf "%{buildroot}"
mkdir -p "%{buildroot}/%{_libdir}"
mkdir -p "%{buildroot}/%{_includedir}"
mkdir -p "%{buildroot}/%{_mandir}"
make V=1 DESTDIR="%{buildroot}" install
rm -f "%{buildroot}/%{_libdir}/libseccomp.la"
rm -f "%{buildroot}/%{_libdir}/libseccomp.a"

%check
make V=1 check

%post -p /sbin/ldconfig

%postun -p /sbin/ldconfig

%files
%{!?_licensedir:%global license %%doc}
%license LICENSE
%doc CREDITS
%doc README
%doc CHANGELOG
%doc SUBMITTING_PATCHES
%{_libdir}/libseccomp.so.*

%files devel
%{_includedir}/seccomp.h
%{_libdir}/libseccomp.so
%{_libdir}/pkgconfig/libseccomp.pc
%{_bindir}/scmp_sys_resolver
%{_mandir}/man1/*
%{_mandir}/man3/*

%changelog
* Wed Feb 22 2017 Paul Moore <pmoore@redhat.com> - 2.3.1-3
- Added the ppc arch to the build

* Thu Apr 28 2016 Paul Moore <pmoore@redhat.com> - 2.3.1-2
- Fix a typo with the ppc64le architecture

* Thu Apr 21 2016 Paul Moore <pmoore@redhat.com> - 2.3.1-1
- Escape the macros in the changelog to make rpmlint and friends happy

* Wed Apr 20 2016 Paul Moore <pmoore@redhat.com> - 2.3.1-0
- New upstream version

* Mon Jun 15 2015 Paul Moore <pmoore@redhat.com> - 2.2.1-1
- Removed '--disable-static' from the build to ensure that scmp_sys_resolver
  is self contained and resolve RPATH issues

* Wed May 13 2015 Paul Moore <pmoore@redhat.com> - 2.2.1-0
- New upstream version
- Added aarch64 support
- Move to an autotools based build system

* Thu Feb 27 2014 Paul Moore <pmoore@redhat.com> - 2.1.1-2
- Build with CFLAGS="${optflags}" (RHBZ #1070774)
* Fri Dec 27 2013 Daniel Mach <dmach@redhat.com> - 2.1.1-1
- Mass rebuild 2013-12-27

* Tue Nov  5 2013 Paul Moore <pmoore@redhat.com> - 2.1.1-0
- New upstream version
- Added a %%check procedure for self-test during build
* Tue Jun 11 2013 Paul Moore <pmoore@redhat.com> - 2.1.0-0
- New upstream version
- Added support for the ARM architecture
- Added the scmp_sys_resolver tool
* Mon Jan 28 2013 Paul Moore <pmoore@redhat.com> - 2.0.0-0
- New upstream version
* Tue Nov 13 2012 Paul Moore <pmoore@redhat.com> - 1.0.1-0
- New upstream version with several important fixes
* Tue Jul 31 2012 Paul Moore <pmoore@redhat.com> - 1.0.0-0
- New upstream version
- Remove verbose build patch as it is no longer needed
- Enable _smp_mflags during build stage
* Thu Jul 19 2012 Fedora Release Engineering <rel-eng@lists.fedoraproject.org> - 0.1.0-2
- Rebuilt for https://fedoraproject.org/wiki/Fedora_18_Mass_Rebuild
* Tue Jul 10 2012 Paul Moore <pmoore@redhat.com> - 0.1.0-1
- Limit package to x86/x86_64 platforms (RHBZ #837888)
* Tue Jun 12 2012 Paul Moore <pmoore@redhat.com> - 0.1.0-0
- Initial versio
```

Thanks for the help!
